indexer/ftraits.hpp

```diff
@@ -190,6 +197,19 @@ class UGC : public TraitsBase<UGC, UGCItem>
       return opt->m_categories;
     return {};
   }
+
+  static boost::optional<std::string> GetMatchedType(UGCRatingCategories const & categories)
+  {
+    auto const & matcher = GetMatcher();
+    auto const end = matcher.End();
+    for (auto it = matcher.Begin(); it != end; ++it)
+    {
+      if (it->second.m_categories == categories)
+        return it->second.m_matchedType;
```

Comment:
Consider adding a comment that explains why we return the first type that matches here.